index.d.ts

```diff
@@ -0,0 +1,21 @@
+declare module "react-native-hyperlinked-text" {
+
+    import {TextStyle} from "react-native";
+
+    export interface ILinkDef {
+        regex: RegExp;
+        style?: TextStyle;
+        replaceText?: (orig: string, text: string, url: string) => string;
+        onPress?: (orig: string, text: string, url: string) => void;
+    }
+
+    interface IProps {
+        style?: TextStyle;
+        linkDefs?: ILinkDef[];
+
```

Comment:
Here we have also `onLinkPress` and `linkStyle` that let you define global press handler and style for links.